runtime/compiler/optimizer/VectorAPIExpansion.hpp

```diff
@@ -264,7 +265,7 @@ class TR_VectorAPIExpansion : public TR::Optimization
    static TR::VectorLength supportedOnPlatform(TR::Compilation *comp, vec_sz_t vectorLength)
          {
          // General check for supported infrastructure
-         if (!comp->target().cpu.isPower() && !comp->target().cpu.isZ() && !comp->target().cpu.isARM64())
+         if (!comp->target().cpu.isPower() && !(comp->target().cpu.isZ() && comp->cg()->getSupportsVectorRegisters()) && !comp->target().cpu.isARM64())
```

Comment:
Maybe split it into 3 lines, one per platform?

Reply:
Changed in https://github.com/eclipse-openj9/openj9/compare/e3075499873373cfd5914d876e03c977c4bec7e8..cb468f4114cd748f7749011a4d476ad5023a4b1a

Reply:
Had a typo in last push, fixed it in https://github.com/eclipse-openj9/openj9/compare/cb468f4114cd748f7749011a4d476ad5023a4b1a..7ea3491e40e4dd1647e00312e1140b330dc66ff2